File: tournaments/visitsOnCircularRoad/visitsOnCircularRoad.cs
int visitsOnCircularRoad(int n, int[] visitsOrder) {
    int current = 1,
        res = 0;
    for (var i = 0; i < visitsOrder.Length; i++) {
        res += Math.Min(Math.Abs(visitsOrder[i] - current),
                    n - Math.Abs(visitsOrder[i] - current));
        current = visitsOrder[i];
    }
    return res;
}

File: tournaments/visitsOnCircularRoad/visitsOnCircularRoad.cpp
int visitsOnCircularRoad(int n, std::vector<int> visitsOrder) {

  int current = 1,
      res = 0;
  for (int i = 0; i < visitsOrder.size(); i++) {
    res += std::min(std::abs(visitsOrder[i] - current),
               n - std::abs(visitsOrder[i] - current));
    current = visitsOrder[i];
  }
  return res;
}
